src/collapse/vsprof.rs

```diff
@@ -0,0 +1,188 @@
+        // causes the call to `starts_with` to return false, which in this case isn't what we want.
+        // As this character has no influence on the rest of the file, we can safely ignore it.
+        if line.starts_with(START_LINE) || line.starts_with(&("\u{feff}".to_owned() + START_LINE)) {
+            Some(true)
+        } else {
+            Some(false)
+        }
+    }
+}
+
+impl Folder {
+    // Parse lines like:
+    // Level,Function Name,Number of Calls,...
+    // 6,"System.String.IsNullOrEmpty(string)",4,0.00,0.00,0.00,0.00,"mscorlib.dll",
+    fn on_line(&mut self, line: &str, occurences: &mut Occurrences) -> io::Result<()> {
+        let (depth, remainder) = get_next_number(line)?;
+        // After the depth is parsed from `line`, it still has a leading comma and double
+        // quote. By trimming those, we know that the first double quote is the closing double
+        // quote surrounding the function name. By splitting on this double quote, we get the
+        // function name and the remainder of the line.
+        let x: &[_] = &[',', '"'];
+        let mut split = remainder.trim_start_matches(x).splitn(2, '"');
+        match (split.next(), split.next()) {
+            (Some(function_name), Some(remainder)) => {
+                let (number_of_calls, _) = get_next_number(remainder)?;
+
+                let prev_depth = self.stack.len();
+                // There are 3 separate cases to handle regarding the depth:
+                // 1. prev_depth + 1 == depth -> a new function is called, we only need to
+                //    store the function name and the number of times it is called from the
+                //    outer function
+                // 2. prev_depth == depth -> the previous function call was a leaf node, so we
+                //    need to save the current stack and replace the top node with our node
+                //    call
+                // 3. prev_depth > depth -> the previous function call was a leaf node, so we
+                //    need to save the current stack and than we need to pop the top nodes
+                //    until the top node is our parent
+                match prev_depth.cmp(&depth) {
+                    // Case 1
+                    Ordering::Less => {
+                        self.stack
+                            .push((function_name.to_string(), number_of_calls));
+                    }
+                    // Case 2
+                    Ordering::Equal => {
+                        self.write_stack(occurences);
+                        self.stack.pop();
+                        self.stack
+                            .push((function_name.to_string(), number_of_calls));
+                    }
+                    // Case 3
+                    Ordering::Greater => {
+                        self.write_stack(occurences);
+
+                        // Pop (prev_depth - depth - 1) nodes of the stack, this leaves our
+                        // parent as the top node.
```

Comment:
Something seems off here. Let's say the stack depth goes from 5 to 3. Presumably I should register the sample counts for both the node at 5 and the one at 4? But I think the single call to `write_stack` above will _just_ register the sample count for the stack at depth 5, but not the one at 4, which then gets popped (and thus forgotten) just below?